### ISSUE TITLE
Type contentRef is not assignable with HTMLDivElement

### ISSUE THREAD

**Issue by shiqocred** [MatthewHerbst/react-to-print#768]
in version ^3.0.2 and react 19 with typescript in nextjs, i cannot use ref with type <HTMLDivElemet | null> in contentRef cause contentRef require type <Element | Text>

<img width="831" alt="Jepretan Layar 2024-12-03 pukul 20 28 50" src="https://github.com/user-attachments/assets/e908f6fb-1577-46c7-a5db-3f6b0065ddc9">

**Comment by MatthewHerbst** [comment]
Hey! Thanks for the report. I'll look into this later today and if I'm able to replicate will push a fix this evening

**Comment by MatthewHerbst** [comment]
I just did a quick test and I am not able to replicate this. One common issue I've seen is folks forgetting to initialize their ref properly:

```tsx
const contentRef = useRef<HTMLDivElement | null>(null); // Initializing the `null` here is important
```

Does that resolve your issue by chance?

**Comment by MatthewHerbst** [comment]
Going to close this assuming the above solved your issue. Please let me know if it did not and we can re-open. Thanks!

**Comment by shiqocred** [comment]
i'm so sorry, i forget for this issue.

I have done as you suggested, but the problem remains the same

<img width="1066" alt="Jepretan Layar 2024-12-06 pukul 17 18 55" src="https://github.com/user-attachments/assets/38cf3ba5-e4ca-4e0b-81bc-ab32652c07ff">

**Comment by shiqocred** [comment]
i have solved the problem with remove `( ) =>` and `.current`

**Comment by MatthewHerbst** [comment]
Glad you got it figured out!

**Comment by marvinroger** [comment]
Hi @MatthewHerbst!

I do have the same issue, with exactly this and React 19:

```typescript
const contentRef = useRef<HTMLDivElement | null>(null);
const print = useReactToPrint({
  contentRef,
})
```

The expected `contentRef` type is `contentRef?: RefObject<Element | Text>`, but `useRef<HTMLDivElement | null>` is `RefObject<HTMLDivElement | null>`, so I don't see how it could work?

The error is 

> Type 'RefObject<HTMLDivElement.
Type 'HTMLDivElement | null' is not assignable to type 'Element | Text'.
Type 'null' is not assignable to type 'Element | Text'.

I'm using `@types/react: 19.02` and `react-to-print: 3.0.2`

**Comment by MatthewHerbst** [comment]
Not sure why I'm not able to replicate this locally. I will look to make the types more friendly though with an `extends`. Need to ensure I avoid making a breaking types change though

**Comment by marvinroger** [comment]
I can consistently reproduce, here is a minimal reproduction: https://stackblitz.com/edit/vitejs-vite-ba4sdiki?file=src%2FApp.tsx

@MatthewHerbst I think you might have some `@types/react@^18` laying around, which get merged with the 19 types. I did not have the problem at first, but deduplicating the peer dependencies (`yarn dedupe` in my case) led to this issue

**Comment by MatthewHerbst** [comment]
Ah, I think you are correct, I can reproduce with a clean local using the React 19 types. Will push out a new version shortly

**Comment by MatthewHerbst** [comment]
To be clear also, the issue is with the `null`. `HTMLDivElement` is an extension of `Element` and thus satisfies the `Element` type

**Comment by MatthewHerbst** [comment]
Fix is published in `v3.0.3`, thank you both for the reports, and please let me know if you have any additional issues
